### PR TITLE
TSK-1166 exploit states TERMINATED and CANCELLED from adapter

### PR DESCRIPTION
**Pull request by BerndBreier**
SonarCloud is green: https://sonarcloud.io/dashboard?branch=TSK-1166&id=BerndBreier_TaskanaAdapter

### CODE REVIEW

File: taskana-adapter-camunda-spring-boot-test/src/test/java/pro/taskana/adapter/integration/TestCancelledTaskRetrieval.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public void deletion_of_camunda_process_instance_should_complete_taskana_task()
       Instant taskanaTaskCompletion = taskanaTasks.get(0).getCompleted();
       Instant taskanaTaskCreation = taskanaTasks.get(0).getCreated();
       TaskState taskanaTaskState = taskanaTasks.get(0).getState();
-      assertThat(TaskState.COMPLETED.equals(taskanaTaskState)).isTrue();
+      assertThat(TaskState.TERMINATED.equals(taskanaTaskState)).isTrue();
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-spring-boot-test/src/test/java/pro/taskana/adapter/integration/TestCancelledTaskRetrieval.java:134]: Can't we use assertThat(taskanaTaskState).isEqualTo(TaskState.TERMINATED) ?

Comment (BerndBreier) [reply taskana-adapter-camunda-spring-boot-test/src/test/java/pro/taskana/adapter/integration/TestCancelledTaskRetrieval.java]: Done

File: taskana-adapter-camunda-listener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java [hunk 2]
```diff
@@ -131,14 +134,22 @@ private void insertCompleteOrDeleteEventIntoOutbox(
...
+      if (delegateTask.getEventName().equals("delete")) {
+        if (delegateTask.getExecution().isCanceled()) {
+          taskState = TASK_STATE_CANCELLED;
+        } else {
+          taskState = TASK_STATE_TERMINATED;
+        }
+      }
 
-      String taskIdJson = "{\"id\":\"" + delegateTask.getId() + "\"}";
+      String referencedTaskJson = String.format("{\"id\":\"%s\",\"taskState\":\"%s\"}",
```
Comment (gitgoodjhe) [inline taskana-adapter-camunda-listener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java:146]: If we don't use objectMapper this is not really a referencedTaskJson anymore. Maybe payloadJson or taskIdAndTaskStateJson is more fitting?

Comment (BerndBreier) [reply taskana-adapter-camunda-listener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java]: I disagree. 
1. It is identical to the string that is produced by objectMapper if you instruct objectmapper not to serialize nulls
2. on the adapter side, this String is deserialized by ObjectMapper into a ReferencedTask.
Nevertheless, I renamed it to 'payload'....

**Pull request merged**